pkg/reconciler/pipeline/controller.go

```diff
@@ -534,6 +534,9 @@ func needsUpdate(old, new *dfv1.Pipeline) bool {
 	if !equality.Semantic.DeepEqual(old.Finalizers, new.Finalizers) {
 		return true
 	}
+	if !equality.Semantic.DeepEqual(old.GetAnnotations(), new.GetAnnotations()) {
```

Comment:
Have a concern of comparing all the annotations might cause unexpected behaviors, maybe just check if the keys starting with `numaflow.numaproj.io/` has any change?